src/libraries/iterableMap.se

```diff
@@ -28,7 +29,20 @@ def addOrUpdate(key, value):
     else:
         self.update(key, value)
 
-# TODO: add support for item removal, note that this requires moving the last item in the collection to the newly opened slot so we don't end up with a sparse collection
+def remove(key):
+    require(self.contains(key))
+    keyRemovedOffset = self.itemsMap[key].offset
+    self.itemsArray[keyRemovedOffset] = 0
+    self.itemsMap[key].hasValue = 0
+    self.itemsMap[key].value = 0
+    self.itemsMap[key].offset = 0
+    if(self.numberOfItems > 1 and keyRemovedOffset != self.numberOfItems):
```

Comment:
Won't the second half of this always be true?  Number of items will be one greater than the offset of the last item, meaning in will never equal offset of removed item.